addons/sms/models/ir_actions.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ServerActions(models.Model):
+    """ Add SMS option in server actions. """
+    _name = 'ir.actions.server'
+    _inherit = ['ir.actions.server']
+
+    state = fields.Selection(selection_add=[
+        ('sms', 'Send SMS Text Message'),
+    ])
+    # SMS
+    sms_template_id = fields.Many2one(
+        'sms.template', 'SMS Template', ondelete='set null',
+        domain="[('model_id', '=', model_id)]",
+    )
+    sms_mass_keep_log = fields.Boolean('Log a note', default=True)
+
+    @api.constrains('state', 'model_id')
+    def _check_sms_capability(self):
+        for action in self:
+            if action.state == 'sms' and not action.model_id.is_mail_thread:
+                raise ValidationError(_("Sending SMS can only be done on a mail.thread model"))
+
+    @api.model
```

Comment:
`api.model` is dead in `master`

Reply:
api.multi is dead api.model is still usefull